scripts/db_migrator.py

```diff
@@ -51,6 +53,12 @@ def __init__(self, namespace, socket=None):
         self.TABLE_KEY       = 'DATABASE'
         self.TABLE_FIELD     = 'VERSION'
 
+        # load config data from minigraph to get the default/hardcoded values from minigraph.py
+        # this is to avoid duplicating the hardcoded these values in db_migrator
+        self.minigraph_data = None
+        if os.path.isfile(MINIGRAPH_FILE):
```

Comment:
>isfile [](http://example.com/codeflow?start=19&length=6)

Could you add a testcase for missing MINIGRAPH_FILE? It would be even better if you can add a testcase for L2 switch mode

Reply:
ack

Reply:
Added a new testcase for missing minigraph. We can discuss what other fields are expected/unexpected in L2 switch's case and add a test for that in future PR. What do you suggest?

Reply:
Add a test for L2 switch in future is okay.

Reply:
As per offline discussion: from unit test level the test for missing minigraph and absence of default entries is sufficient. L2 switch config and functionality needs to be tested separately as part of sonic-mgmt test. Raised a test gap for this: https://github.com/sonic-net/sonic-mgmt/issues/8777